AllASPFuncs.py

```diff
@@ -1292,3 +1307,34 @@ def banner(text):
     length = len(text) + 8
     message = "\n" + "#" * length +'\n'+'#   ' + text + '   # \n'+ "#" * length
     Lager.debug(message)
+
+
+def prep_data_for_fit(images, err, sn_matrix, wgt_matrix):
+    '''
+    This function takes the data from the images and puts it into the form such
+    that we can analytically solve for the best fit using linear algebra.
+    '''
+    size = int(np.sqrt((images[0].size)))
```

Comment:
You only use size_sq below, so I would just calculate size_sq.  Or even perhaps num_pixels.

I bring this up both because 
a. it's simpler
b. but also you're relying here that a float sqrt operation followed by an int squaring operation will be reversible, but that's not guaranteed to work.  This happens to work in Python and numpy if N is a square, but is not guaranteed.
c. The reason that this caught my eye was that I often write code to do something simple like make a tiling of plots, where N may not be a square.  There I want `size = int(sqrt(N)) + 1` to make sure I have a square that includes all of the N.

a = int(sqrt(a*a))